### PR TITLE
Implement the first half of the nmod:desc change... maybe?

### PR DESCRIPTION
**Pull request by AngledLuffa**
https://github.com/UniversalDependencies/UD_English-PUD/issues/60

@nschneid does this look correct?  if so, i will go back and do the second half as well

### CODE REVIEW

File: en_pud-ud-test.conllu [hunk 1]
```diff
@@ -6398,13 +6398,13 @@
 6	,	,	PUNCT	,	_	3	punct	3:punct	_
 7	focusing	focus	VERB	VBG	VerbForm=Ger	2	acl	2:acl	_
 8	on	on	ADP	IN	_	9	case	9:case	_
```
Comment (nschneid) [inline en_pud-ud-test.conllu:6400]: preposition head should also be updated

File: en_pud-ud-test.conllu [hunk 2]
```diff
@@ -6398,13 +6398,13 @@
...
 11	and	and	CCONJ	CC	_	13	cc	13:cc	_
 12	brand	brand	NOUN	NN	Number=Sing	13	compound	13:compound	_
-13	chief	chief	NOUN	NN	Number=Sing	10	conj	9:flat|10:conj:and|18:nsubj	_
-14	Herbert	Herbert	PROPN	NNP	Number=Sing	13	flat	13:flat	_
-15	Diess	Diess	PROPN	NNP	Number=Sing	13	flat	13:flat	SpaceAfter=No
+13	chief	chief	NOUN	NN	Number=Sing	14	nmod:desc	14:nmod:desc	_
+14	Herbert	Herbert	PROPN	NNP	Number=Sing	10	conj	10:conj:and|18:nsubj	_
+15	Diess	Diess	PROPN	NNP	Number=Sing	14	flat	14:flat	SpaceAfter=No
 16	,	,	PUNCT	,	_	18	punct	18:punct	_
 17	who	who	PRON	WP	PronType=Rel	18	nsubj	13:ref	_
```
Comment (nschneid) [inline en_pud-ud-test.conllu:6409]: `ref` head should be updated

File: en_pud-ud-test.conllu [hunk 3]
```diff
@@ -9583,8 +9583,8 @@
 13	Rastiz	Rastiz	PROPN	NNP	Number=Sing	12	appos	12:appos	_
 14	who	who	PRON	WP	PronType=Rel	15	nsubj	12:ref	_
 15	became	become	VERB	VBD	Mood=Ind|Tense=Past|VerbForm=Fin	12	acl:relcl	12:acl:relcl	_
-16	St.	St.	PROPN	NNP	Number=Sing	15	xcomp	15:xcomp	_
-17	Rastislav	Rastislav	PROPN	NNP	Number=Sing	16	flat	16:flat	SpaceAfter=No
```
Comment (nschneid) [inline en_pud-ud-test.conllu:9587]: For "nephew" (token 12), `16:nsubj:xsubj` -> `17:nsubj:xsubj`

**Review by nschneid — CHANGES_REQUESTED** [review]
Updates to title+name expression look good, remember to also update dependents elsewhere in the sentence (including enhanced graph dependents)

**Comment by AngledLuffa** [comment]
Sounds good.  I also found one other, in `expectant mother`

https://github.com/UniversalDependencies/UD_English-PUD/pull/61/commits/e4a0deb954f0bbfb3d08c89d143be36d81ee93e7

**Review by nschneid — APPROVED** [review]

File: en_pud-ud-test.conllu [hunk 4]
```diff
@@ -15879,14 +15879,14 @@
...
 8	was	be	AUX	VBD	Mood=Ind|Number=Sing|Person=3|Tense=Past|VerbForm=Fin	9	aux:pass	9:aux:pass	_
 9	visited	visit	VERB	VBN	Tense=Past|VerbForm=Part	0	root	0:root	_
-10	by	by	ADP	IN	_	11	case	11:case	_
-11	Reverend	Reverend	PROPN	NNP	Number=Sing	9	obl	9:obl:by	_
-12	William	William	PROPN	NNP	Number=Sing	11	flat	11:flat	_
-13	Hechler	Hechler	PROPN	NNP	Number=Sing	11	flat	11:flat	SpaceAfter=No
-14	,	,	PUNCT	,	_	11	punct	11:punct	_
+10	by	by	ADP	IN	_	12	case	12:case	_
+11	Reverend	Reverend	PROPN	NNP	Number=Sing	12	nmod:desc	12:nmod:desc	_
+12	William	William	PROPN	NNP	Number=Sing	9	obl	9:obl:by	_
```
Comment (nschneid) [inline en_pud-ud-test.conllu:15884]: Technically I think this should be `obl:agent` (passive by-phrase) but maybe that's for a separate issue

Comment (nschneid) [reply en_pud-ud-test.conllu:15884]: And I'm realizing this is not yet in the English docs, oops...will work on that.

File: en_pud-ud-test.conllu [hunk 5]
```diff
@@ -16117,8 +16117,8 @@
 6	sought	seek	VERB	VBD	Mood=Ind|Tense=Past|VerbForm=Fin	0	root	0:root	_
 7	for	for	ADP	IN	_	10	case	10:case	_
 8	outgoing	outgoing	ADJ	JJ	Degree=Pos	9	amod	9:amod	_
-9	President	President	PROPN	NNP	Number=Sing	6	obl	6:obl	_
-10	Yudhoyono	Yudhoyono	PROPN	NNP	Number=Sing	9	flat	9:flat	_
+9	President	President	PROPN	NNP	Number=Sing	10	nmod:desc	10:nmod:desc	_
+10	Yudhoyono	Yudhoyono	PROPN	NNP	Number=Sing	6	obl	6:obl	_
```
Comment (nschneid) [inline en_pud-ud-test.conllu:16121]: `obl:for`? Although I'm not sure this is the right analysis. It looks like *for* marking the subject of an infinitival clause.

File: en_pud-ud-test.conllu [hunk 6]
```diff
@@ -16734,8 +16734,8 @@
 3	,	,	PUNCT	,	_	16	punct	16:punct	_
 4	his	his	PRON	PRP$	Gender=Masc|Number=Sing|Person=3|Poss=Yes|PronType=Prs	5	nmod:poss	5:nmod:poss	_
 5	brother	brother	NOUN	NN	Number=Sing	16	nsubj:pass	16:nsubj:pass	_
-6	Prince	prince	NOUN	NN	Number=Sing	5	appos	5:appos	_
-7	Henry	Henry	PROPN	NNP	Number=Sing	6	flat	6:flat	SpaceAfter=No
+6	Prince	prince	NOUN	NN	Number=Sing	6	nmod:desc	6:nmod:desc	_
+7	Henry	Henry	PROPN	NNP	Number=Sing	5	appos	5:appos	SpaceAfter=No
 8	,	,	PUNCT	,	_	9	punct	9:punct	_
 9	Duke	Duke	PROPN	NNP	Number=Sing	5	appos	5:appos	_
```
Comment (nschneid) [inline en_pud-ud-test.conllu:16740]: Probably more natural to attach this to "Prince Henry" i.e. `7:appos`

File: en_pud-ud-test.conllu (hunk 4 above)
Comment (nschneid) [reply en_pud-ud-test.conllu:15884]: https://github.com/UniversalDependencies/docs/pull/1074

File: en_pud-ud-test.conllu (hunk 5 above)
Comment (AngledLuffa) [reply en_pud-ud-test.conllu:16121]: Would you be up for editing that tree?  I'm more of a remote terminal than a processor when it comes to annotating dependencies

File: en_pud-ud-test.conllu (hunk 4 above)
Comment (AngledLuffa) [reply en_pud-ud-test.conllu:15884]: So as a general rule, we should turn passive voice `obl` into `obl:agent`?  I think that might be a separate issue.  There's bound to be quite a few in this treebank

File: en_pud-ud-test.conllu (hunk 6 above)
Comment (AngledLuffa) [reply en_pud-ud-test.conllu:16740]: Done

File: en_pud-ud-test.conllu (hunk 5 above)
Comment (nschneid) [reply en_pud-ud-test.conllu:16121]: VERB + for-X-to-Y is something of a rare pattern. Updated modeled after this one from EWT:
image

File: en_pud-ud-test.conllu (hunk 4 above)
Comment (nschneid) [reply en_pud-ud-test.conllu:15884]: #63

**Comment by nschneid** [comment]
LGTM

**Comment by AngledLuffa** [comment]
Excellent, so, I'll squash & merge?

**Comment by nschneid** [comment]
Sure

**Comment by AngledLuffa** [comment]
Thank you, I will do that now, then

**Comment by nschneid** [comment]
The validator caught a couple errors: http://quest.ms.mff.cuni.cz/udvalidator/cgi-bin/unidep/validation-report.pl?UD_English-PUD

**Comment by AngledLuffa** [comment]
Oops, that was careless of me.  Thanks for pointing those out.

https://github.com/UniversalDependencies/UD_English-PUD/commit/27210d65156615faead42e98ea0b794b94067e76

Speaking of the validator, worth figuring out some of the other ExtPos tags which aren't there yet?

**Comment by nschneid** [comment]
Sure, do you want to take a crack at it?

**Comment by AngledLuffa** [comment]
not really but i guess no one else will

**Comment by AngledLuffa** [comment]
there's also ParTUT, which has several such fixed expressions... does anyone have ownership of that?

**Comment by nschneid** [comment]
> does anyone have ownership of that?

@msang appears to be the maintainer

**Pull request merged**